### PR TITLE
Feature actdisk

### PR DESCRIPTION
**Pull request by fpalacios**
Complete update of the engine and actuator disk boundary conditions, including a new regression test (propeller.cfg).

### CODE REVIEW

**Comment by economon** [comment]
Thanks for submitting this, Francisco. It's going to be a great contribution. I'll need a bit of time to go through it all, but I can review and post here before long.

**Comment by economon** [comment]
@fpalacios: just let me know when you've got everything you want in this pull request. I'll wait to review until then.

**Comment by fpalacios** [comment]
The implementation is ready to be reviewed. Apart form the engine and actuator disk update, the new implementation includes a particular marker called MARKER_INTERNAL. I'm using this MARKER to define transparent internal surfaces inside SU2, the purpose is to define and analyze (average quantities, distortion parameters, etc) these surfaces without affecting time evaluation, or dissipation. Remember that the definition of a typical surface in SU2 affects the way the time step is evaluated or the dissipation in a JST scheme (not if you use MAKER_INTERNAL). As a side note, the actuator disk implementation is based on a double defined surface system but, the surface splitting is done inside SU2.

File: Common/include/config_structure.hpp [hunk 1]
```diff
@@ -4972,197 +5153,961 @@ class CConfig {
...
   /*!
    * \brief Get the back pressure (static) at an outlet boundary.
    * \param[in] val_index - Index corresponding to the outlet boundary.
    * \return The outlet pressure.
    */
-  void SetBleed_MassFlow(unsigned short val_imarker, su2double val_bleed_massflow);
+  void SetInflow_Pressure(unsigned short val_imarker, su2double val_fanface_pressure);
   
   /*!
    * \brief Get the back pressure (static) at an outlet boundary.
```
Comment (economon) [inline Common/include/config_structure.hpp]: Can you please update the oxygen comments for these getter/setter methods? It seems they are all copy/paste of an old one so far...

File: SU2_CFD/src/output_structure.cpp [hunk 2]
```diff
@@ -8883,6 +8892,920 @@ void COutput::SetEquivalentArea(CSolver *solver_container, CGeometry *geometry,
   
 }
 
+void COutput::WriteSurface_Analysis(CConfig *config, CGeometry *geometry, CSolver *FlowSolver) {
```
Comment (economon) [inline SU2_CFD/src/output_structure.cpp]: I think this is a great new capability to write out info on surfaces... it looks like they have to be defined as MARKERS within our grid format, correct? We've been thinking about reusing the interpolation routines and fast searches for collecting data at arbitrary surfaces in the solution. Maybe we should discuss how to unify these efforts.

Comment (talbring) [inline SU2_CFD/src/output_structure.cpp]: A question related to that: is it still necessary to define a double sided boundary for this marker ? We thought of maybe using this implementation for the FWH solver until we have the general extraction (which apparently is more complicated than I thought originally, @economon we have to talk about this soon).

File: Common/include/config_structure.hpp (hunk 1 above)
Comment (fpalacios) [inline Common/include/config_structure.hpp]: Thanks Tom, You are completely right. I'm working on the copy/paste comments, I have already some of them (the most critical for the engine bc) and step by step I'm planning to update all of them. For the time being, I don't have a good example for traditional engine boundary condition, the ideal scenery would be to find an open nacelle geometry (any idea?) to offer a meaningful nacelle with meaningful boundary values. I'll check if the NASA rotor 37 has a nacelle.

File: SU2_CFD/src/output_structure.cpp (hunk 2 above)
Comment (fpalacios) [inline SU2_CFD/src/output_structure.cpp]: The MARKER_INTERNAL is a way to define a surface that does not participate on numerical algorithms (time stepping and dissipation) It took me some time to verify that you have exactly the same solution with or without MARKER_INTERNAL. Then I'm using MARKER_ANALYZE to identify which surfaces I want to analyze in detail (average quantities, distortion, etc) It doesn't need to be an internal surface... in fact, we typically analyze the distortion at the fan face which is an inflow BC. Something that you have noticed is that there are several problems in which you need to project the unstructured grid into a structured grid (distortion analysis, interpolation, interaction with other softwares, etc) I'm implementing that kind of projection in a rudimentary way that we should implement and unify in the future.

After several test, we determined that the double surface is the best method to implement an actuator disk (source terms doesn't work OK with unstructured grids... and it has some problems even for structured solvers). SU2 splits the surface internally, so, from the point of view of the user, the input is an internal surface that SU2 transforms into a double side boundary... it is not a very complicated algorithm.

Comment (economon) [inline SU2_CFD/src/output_structure.cpp]: Yes.. we can try using the MARKER_INTERNAL for FWH. In fact, @sravya91 will likely give this a try asap. Everything looks good (thanks for the updates, Francisco). We'll merge this in asap so we can try out the surfaces.

File: Common/include/config_structure.hpp (hunk 1 above)
Comment (cpt-ahab) [reply Common/include/config_structure.hpp]: Did you ever find the time to try out a test case for the engine BC? I would love to use the feature, but I'm having a hard time interpreting some of the required input in the config file. Thank you!
